### DISCUSSION TITLE
[Visual Debugger] Should positions be shown instead of origins?

### DISCUSSION THREAD

**Discussion by D8H** [4ian/GDevelop#3395] [Enhancement ideas (and their technical discussions)]
The visual debugger shows the origin point. I think that this point doesn't have any meaning for the events.
The events use the position of the object. This is not the same thing as the origin as the position doesn't change when the object rotate.

This short video shows an object rotating with the position instead of the origin:
https://user-images.githubusercontent.com/2611977/147014245-d3cffe37-e27f-43e3-8d4a-6892613fa429.mp4

I think that showing the position is more useful and allows to better understand how `X()` and `Y()` work.

**Comment by Silver-Streak** [comment]
I'm not sure I understand the ask/improvement here. 

As far as I know, everything (other than hitboxes or sprites) uses the origin point as the logical source of truth for location. When you position an object by events, it doesn't use AABB, it uses the origin point. 

Flipping happens on center, but rotation uses origin on every test I can do in the web demo.

Whatsmore, origin point being a source of truth is important as it allows users to move the sprite rendering position without adding more empty space to the art asset by moving the point in yhe points editor.

I would look for someone like @4ian or one of the crew to chime in more incase I'm totally off base and missing something.

**Reply by D8H** [reply]
Actually the difference between the object position and the origin point is that the position doesn't rotate withe the object.
In short, the object position is the "unrotated" origin point.

For instance, when an object rotate, its position stays the same even if the origin is not at the center.

The debugger doesn't show any point for Object.X() and Object.Y().

**Reply by Silver-Streak** [reply]
Sorry, you're correct. I did some testing, and while I agree it rotates around center point/located separately from the origin point, I'm still not sure I understand what you are asking for. What are you calling the "visual debugger"?

Do you mean debug preview? because that shows origin (logical position of the object as currently placed in the scene editor/via events, not linked to origin point as I originally thought) and AABB positions.
![image](https://user-images.githubusercontent.com/24237149/147371488-cd6e4bad-27a8-4ee5-886a-4b66a8fce1a7.png)

I'm not sure what you're asking for (by "positions") different than the currently described x/y attribute that's already in the debugger.
(I'd actually ask that the debugger show point locations as a parameter, since it doesn't today. )

**Reply by D8H** [reply]
I meant what this action shows:
![image](https://user-images.githubusercontent.com/2611977/147371609-4b5ad730-52e2-4e10-bd7c-cf3f821b6b64.png)

**Comment by tristanbob** [comment]
I just learned what you talk about in this post; Object.X() and Object.Y() don't move when the object is rotated.  This is highly unexpected. 

I think we should add new points called Object.OriginX() and Object.OriginY() that do move when the object is rotated. 

What do you think?

**Reply by D8H** [reply]
The origin point can be retrieved with Sprite.PointX("Origin"). Even though the autocompletion doesn't suggest it.

**Reply by tristanbob** [reply]
Thanks, I will use that in the future.  

Do you think it should it be suggested by auto-complete?
